### PR TITLE
GraphQL BTreeMap

### PR DESCRIPTION
**Pull request by maxtori**
## Motivation

The GraphQL server fails to serialize some operations due to the presence of maps indexed by something other than strings (Epoch, ValidatorName).
cf [async-graphql](https://github.com/async-graphql/async-graphql/blob/master/value/src/serializer.rs#L439-L640)

## Proposal

Implement Serialize and Deserialize on committees and validators BTreeMap to change the keys to string.

## Test Plan

- Checking GraphiQL for the block 0 after a run_local.sh. The OpenChain operation should be visible and not `null` as before this PR
- Adding an OpenChain operation in tests in #975 PR

## Release Plan

- [ ] All good!
- [x] Need to bump the major/minor version number in the next release of the crates.
- [ ] Need to update the developer manual.
- [ ] This PR is adding or removing Cargo features.
- [ ] Release is blocked and/or tracked by other issues (see links above)

## Reviewer Checklist

- [ ] The title and the summary of the PR are short and descriptive.
- [ ] The proposed solution achieves the goals stated in the PR.
- [ ] The test plan is reproducible and provides sufficient coverage.
- [ ] The release plan is adequate.
- [ ] The commits correspond to distinct logical changes.
- [ ] The code follows the [coding guidelines](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md).
- [ ] The proposed changes look correct.
- [ ] The CI is passing.
- [ ] All of the above!

### CODE REVIEW

File: linera-execution/src/committee.rs [hunk 1]
```diff
@@ -30,11 +31,46 @@ pub struct ValidatorState {
     pub votes: u64,
 }
 
+#[derive(Clone, Debug, Default, Eq, Hash, PartialEq)]
+pub struct Validators(pub BTreeMap<ValidatorName, ValidatorState>);
```
Comment (ma2bd) [inline linera-execution/src/committee.rs:35]: Here, we should actually just implement `Serialize` and `Deserialize` for `ValidatorName` following the model of `PublicKey` in `crypto.rs`.

File: linera-execution/src/system.rs [hunk 2]
```diff
@@ -35,6 +35,39 @@ use thiserror::Error;
 #[cfg(any(test, feature = "test"))]
 use {crate::applications::ApplicationRegistry, std::collections::BTreeSet};
 
+#[derive(Clone, Debug, Default, Eq, Hash, PartialEq)]
+pub struct Committees(pub BTreeMap<Epoch, Committee>);
```
Comment (ma2bd) [inline linera-execution/src/system.rs:39]: Same here: just (de)serialize `Epoch` as a `u64` in the case of readable formats.

**Review by ma2bd — CHANGES_REQUESTED** [review]
Thanks for looking into this. We can actually use the `is_human_readable` API to avoid introducing new struct types.

File: linera-execution/src/system.rs (hunk 2 above)
Comment (ma2bd) [reply linera-execution/src/system.rs:39]: Something like that:
```
impl Serialize for Epoch {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: serde::ser::Serializer,
    {
        if serializer.is_human_readable() {
            serializer.serialize_u64(&self.0)
        } else {
            serializer.serialize_newtype_struct("Epoch", &self.0)
        }
    }
}

impl<'de> Deserialize<'de> for Epoch {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: serde::de::Deserializer<'de>,
    {
        if deserializer.is_human_readable() {
            let s = u64::deserialize(deserializer)?;
            let value = Self::from_str(&s).map_err(serde::de::Error::custom)?;
            Ok(value)
        } else {
            #[derive(Deserialize)]
            #[serde(rename = "Epoch")]
            struct E(u64);

            let value = E::deserialize(deserializer)?;
            Ok(Self(value.0))
        }
    }
}
```

**Comment by maxtori** [comment]
Unfortunately this doesn't solve the issue. I think the async-graphql scalar macro is checking the type of the key of the map without checking if it can be serialized in a human readable string, so the serialization needs to be modified at the BTreeMap level.
I will try to incorporate your changes so that it doesn't fail on the tests

**Comment by maxtori** [comment]
Ah no my bad, I just need to use `serializer.serialize_str(&self.0.to_string())` for the Epoch.
I will close this PR and open an other one with your changes

**Pull request closed without merging**